improver/developer_tools/metadata_interpreter.py

```diff
@@ -0,0 +1,563 @@
+                cube.name()
+            )
+        except ValueError as cause:
+            # if the probability name is not valid
+            self.errors.append(str(cause))
+
+        expected_threshold_name = get_threshold_coord_name_from_probability_name(
+            cube.name()
+        )
+
+        if not cube.coords(expected_threshold_name):
+            msg = f"Cube does not have expected threshold coord '{expected_threshold_name}'; "
+            try:
+                threshold_name = find_threshold_coordinate(cube).name()
+            except CoordinateNotFoundError:
+                coords = [coord.name() for coord in cube.coords()]
+                msg = (
+                    f"no coord with var_name='threshold' found in all coords: {coords}"
+                )
+                self.errors.append(msg)
+            else:
+                msg += f"threshold coord has incorrect name '{threshold_name}'"
+                self.errors.append(msg)
+                self.check_threshold_coordinate_properties(
+                    cube.name(), cube.coord(threshold_name)
+                )
+        else:
+            threshold_coord = cube.coord(expected_threshold_name)
+            self.check_threshold_coordinate_properties(cube.name(), threshold_coord)
+
+    def check_threshold_coordinate_properties(self, cube_name, threshold_coord):
+        """Checks threshold coordinate properties are correct and consistent with
+        cube name"""
+        threshold_var_name = threshold_coord.var_name
+        if threshold_var_name != "threshold":
+            self.errors.append(
+                f"Threshold coord {threshold_coord.name()} does not have "
+                "var_name='threshold'"
+            )
+
+        self.relative_to_threshold = threshold_coord.attributes[
+            "spp__relative_to_threshold"
+        ]
+
+        if self.relative_to_threshold in ("greater_than", "greater_than_or_equal_to"):
+            threshold_attribute = "above"
+        elif self.relative_to_threshold in ("less_than", "less_than_or_equal_to"):
+            threshold_attribute = "below"
+        elif self.relative_to_threshold == "between_thresholds":
+            threshold_attribute = "between"
+            self.warnings.append("Between thresholds data are not fully supported")
+        else:
+            threshold_attribute = None
+            self.errors.append(
+                f"spp__relative_to_threshold attribute '{self.relative_to_threshold}' "
+                "is not in permitted value set"
+            )
+
+        if threshold_attribute and threshold_attribute not in cube_name:
+            self.errors.append(
+                f"Cube name '{cube_name}' is not consistent with "
+                f"spp__relative_to_threshold attribute '{self.relative_to_threshold}'"
+            )
+
+    def check_cell_methods(self, cube):
+        """Checks cell methods are permitted and correct"""
+        cell_methods = cube.cell_methods
```

Comment:
"cell_methods" is only used once, on the next line. Suggest removing it.

Reply:
OK